com.unity.sg2/Tests/HeadlessPreview/AllNodesTest.cs

```diff
@@ -0,0 +1,134 @@
+                            break;
+
+                        case BaseTextureType.TextureType.Texture3D:
+                            var key3d = new RegistryKey { Name = "SampleTexture3D", Version = 1 };
+                            Graph.AddNode(key3d, previewName);
+                            Graph.TryConnect(nodeName, portName, previewName, "Texture");
+                            break;
+
+                        case BaseTextureType.TextureType.Texture2DArray:
+                            var keyArray = new RegistryKey { Name = "SampleTexture2DArray", Version = 1 };
+                            Graph.AddNode(keyArray, previewName);
+                            Graph.TryConnect(nodeName, portName, previewName, "TextureArray");
+                            break;
+
+                        case BaseTextureType.TextureType.CubeMap:
+                            var keyCube = new RegistryKey { Name = "SampleReflectedCubemap", Version = 1 };
+                            Graph.AddNode(keyCube, previewName);
+                            Graph.TryConnect(nodeName, portName, previewName, "Cube");
+                            break;
+                    }
+                }
+                if (typeName == SamplerStateType.kRegistryKey.Name)
+                {
+                    var key2d = new RegistryKey { Name = "SampleTexture2D", Version = 1 };
+                    Graph.AddNode(key2d, previewName);
+                    Graph.TryConnect(nodeName, portName, previewName, "Sampler");
+                }
+                if (typeName == GradientType.kRegistryKey.Name)
+                {
+                    Graph.AddNode<SampleGradientNode>(previewName);
+                    Graph.TryConnect(nodeName, outPort.LocalID, previewName, SampleGradientNode.kGradient);
+                }
+            }
+
+
+            // The following would be the prefered output, as we could get a much more concise idea of what is wrong and how to fix it--
+            // but the Shader Compiler floods the test results window before ShaderMessages can be accessed, meaning any additional outputs
+            // would end up exceeding the 15000 character limit (and appear at the bottom). It's possible to get this to work correctly
+            // by not routing through the PreviewManager, but the setup and behavior of nodes would then be inconsistent with where this
+            // is relevant.
+
+            //Preview.RequestNodePreviewShaderCodeStrings(previewName, out var shaderMessages, out _, out var prevCode, out _);
+            //string dump = "";
+            //foreach (var msg in shaderMessages)
+            //    dump += msg + "\n";
+            //dump += prevCode;
+            //Assert.IsNotEmpty(shaderMessages, dump);
+
+            var material = Preview.RequestNodePreviewMaterial(previewName);
+            var value = PreviewTestFixture.SampleMaterialColor(material);
+            Assert.AreNotEqual(new Color(1,1,0,1), value);
```

Comment:
Can we make the `Color` for this test `static readonly` and named for what it is?